pkg/xray/watches.go

```diff
@@ -0,0 +1,246 @@
+package xray
+
+import (
+	"context"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"log"
+	"net/http"
+
+	"github.com/go-resty/resty/v2"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+type WatchGeneralData struct {
+	Name        string `json:"name"`
+	Description string `json:"description,omitempty"`
+	Active      bool   `json:"active,omitempty"`
+}
+
+type WatchFilterValue struct {
+	Key   string `json:"key,omitempty"`
+	Value string `json:"value,omitempty"`
+}
+
+type WatchFilter struct {
+	Type  string `json:"type,omitempty"`
+	Value string `json:"value,omitempty"`
+}
+
+type WatchProjectResource struct {
+	Type            string        `json:"type,omitempty"`
+	BinaryManagerId string        `json:"bin_mgr_id,omitempty"`
+	Filters         []WatchFilter `json:"filters,omitempty"`
+	// Watch a repo
+	Name string `json:"name,omitempty"`
+}
+
+type WatchProjectResources struct {
+	Resources []WatchProjectResource `json:"resources,omitempty"`
+}
+
+type WatchAssignedPolicy struct {
+	Name string `json:"name,omitempty"`
+	Type string `json:"type,omitempty"`
+}
+
+type Watch struct {
+	GeneralData      WatchGeneralData      `json:"general_data,omitempty"`
+	ProjectResources WatchProjectResources `json:"project_resources,omitempty"`
+	AssignedPolicies []WatchAssignedPolicy `json:"assigned_policies,omitempty"`
+}
+
+func unpackWatch(d *schema.ResourceData) *Watch {
+	watch := new(Watch)
+
+	gd := WatchGeneralData{
+		Name: d.Get("name").(string),
```

Comment:
Why not calls `d.GetStr()` here?

Reply:
is that available on `d`? That's definitely preferred

Reply:
no, d doesn't have `GetStr()` Jost this: 
![image](https://user-images.githubusercontent.com/55418063/144913329-be98d686-b10b-4797-a6a2-1b45361fa0d7.png)

Reply:
If you create a `ResourceData` (note, *not* `schema.ResourceData`) like:

```go
d := &ResourceData{data}
```

Then you get `d.getString()`

Reply:
Not sure if it solves anything.

Reply:
I've tried to do it (in Projects as well): `Cannot use 'createProject' (type func(ctx context.Context, data ResourceData, m interface{}) diag.Diagnostics) as the type CreateContextFunc`
None of the CRUD functions won't work with the methods with that signature.

Reply:
thanks! done